docs/GSoC.md

```diff
@@ -0,0 +1,80 @@
+
+There are many organizations which generate a weekly report of all the events occurring in their organizations. However, their weekly reports or weekly digests are manually generated or are generated by external bots, and these bots are only organization depended and they need to be configured per organization or per-repository basis. Also, these external bots are not First Actors in GitHub.
+
+It can be seen that users or organizations are highly interested in maintaining a weekly digest for their repository, however, due to the lack of an application, which does this task, they have to do it manually or using some external application.
+
+## The Solution
+
+To solve this problem, I developed the Weekly Digest app based on Probot Framework which generates a weekly automated summary of activity on your GitHub repository.
+
+Th app listens to [Github Webhook Events](https://developer.github.com/webhooks/#events) like `installation.created` and `installation_repositories`, and also to `schedule.repository` which are the GitHub events that trigger the application.
+
+The app then reads the configuration of the application (if it's available) and checks the details that it has to publish using [probot-config](https://www.npmjs.com/package/probot-config).
+
+Then the application gets details like issues, pull requests, commits, contributors, stargazers and releases for the repositories (Using [Octokit REST APIs](https://octokit.github.io/rest.js/)) and adds [GitHub Markdown](https://guides.github.com/features/mastering-markdown/) to it.
+
+Finally, the application publishes the fetched details an issue and adds _weekly-digest_ label to it. This is repeated every week, by default on Sunday.
+
+### So, what the app does?
+
+When the Weekly Digest App gets installed in a repository of a user or organization, it curates together the following data and publishes it as an issue:
+
+- Issues created in the last one week
+  - Open Issues
+  - Closed Issues
+  - Noisy Issue
+  - Liked Issue
+- Pull requests opened, updated, or merged in the last pull request
+  - Opened Pull Requests
+  - Updated Pull Requests
+  - Merged Pull Requests
+- Commits made in the master branch, in the last week
+- Contributors, adding contributions in the last week
+- Stargazers, or the fans of your repositories, who really loved your repo
+- Releases, of the project you are working on
+
+The app, as the name suggests, generates these digests and publishes it on weekly basis, typically on a Sunday. 
+
+One can change the default configuration of the app by adding a _.github/weekly-digest.yml_ file in their GitHub repository, which allows them to configure the publish day and the specific information included in the digest.
+
+## What I learned?
+  - Developing GitHub apps
+  - Using GitHub v3 APIs
+  - Using Octokit REST APIs
+  - Adding Tests using Jest
+  - Developing app triggered by webhooks
```

Comment:
If this is your first Open Source project, you can also add that to your learnings :) Like "creating & maintaining an open source project"

Reply:
Oh and I don’t know about you, but I sure learned a lot about the usage of Emojis from the Probot Community :D